react/features/screenshot-capture/CameraCaptureDialog.tsx

```diff
@@ -0,0 +1,133 @@
+import React, { useCallback, useMemo } from 'react';
+import { WithTranslation } from 'react-i18next';
+import { connect, useDispatch } from 'react-redux';
+import { makeStyles } from 'tss-react/mui';
+
+import { hideDialog } from '../base/dialog/actions';
+import { isMobileBrowser } from '../base/environment/utils';
+import { translate } from '../base/i18n/functions';
+import { withPixelLineHeight } from '../base/styles/functions.web';
+import Dialog from '../base/ui/components/web/Dialog';
+
+const FILE_INPUT_ID = 'captureCameraPictureInput';
+
+const useStyles = makeStyles()(theme => {
+    return {
+        container: {
+            display: 'flex',
+            height: '100%',
+            width: '100%',
+            justifyContent: 'center',
+            alignItems: 'center'
+        },
+
+        hidden: {
+            display: 'none'
+        },
+        label: {
+            background: theme.palette.action02,
+            color: theme.palette.text04,
+            ...withPixelLineHeight(theme.typography.bodyLongBold),
+            borderRadius: theme.shape.borderRadius,
+            boxSizing: 'border-box',
+            cursor: 'pointer',
+            display: 'inline-block',
+            padding: '7px 16px',
+            position: 'relative' as const,
+            textAlign: 'center',
+            border: 0,
+
+            '&:hover': {
+                backgroundColor: theme.palette.action02Hover
+            }
+        },
+        selectContainer: {
+            marginBottom: '8px'
+        }
+    };
+});
+
+/**
+ * The type of {@link CameraCaptureDialog}'s React {@code Component} props.
+ */
+interface IProps extends WithTranslation {
+    /**
+     * Callback function on file input changed.
+     */
+    callback: ({ error, dataURL }: { dataURL?: string; error?: string; }) => void;
+
+    /**
+     * Camera facing mode (environment/user).
+     */
+    cameraFacingMode?: string;
+}
+
+/**
+ * Implements the Camera capture dialog.
+ *
+ * @param {Object} props - The props of the component.
+ * @returns {React$Element}
+ */
+const CameraCaptureDialog = ({ callback, t, cameraFacingMode }: IProps) => {
+    const dispatch = useDispatch();
+    const { classes } = useStyles();
+    const isMobile = isMobileBrowser();
```

Comment:
Why are we checking for isMobile, I'd say the API should return an error in a non-mobile browser case, since the API is not really supported there, is it?